tests/test_buffer_traditional.py

```diff
@@ -138,6 +141,30 @@ def test_zero_cable_len_profile_update(self, dvs, setup_teardown_test):
                 fvs["SAI_INGRESS_PRIORITY_GROUP_ATTR_BUFFER_PROFILE"] = self.buf_pg_profile[pg]
                 self.asic_db.wait_for_field_match("ASIC_STATE:SAI_OBJECT_TYPE_INGRESS_PRIORITY_GROUP", self.pg_name_map[pg], fvs)
 
+            # Verify pgs are updated if pfc_enable is updated
+            orig_lossless_pgs = self.lossless_pgs
+            orig_pg_name_map = self.pg_name_map
+            
+            self.set_port_qos_table('0,2,3,7')
+            self.pg_name_map = self.get_pg_name_map()
+            time.sleep(1)
+            for pg in orig_lossless_pgs:
+                if pg not in self.lossless_pgs:
+                    assert len(self.app_db.get_entry("BUFFER_PG_TABLE", self.INTF + ":" + pg)) == 0
+            for pg in orig_pg_name_map:
+                if pg not in self.pg_name_map:
+                    fvs["SAI_INGRESS_PRIORITY_GROUP_ATTR_BUFFER_PROFILE"] = "oid:0x0"
+                    self.asic_db.wait_for_field_match("ASIC_STATE:SAI_OBJECT_TYPE_INGRESS_PRIORITY_GROUP", orig_pg_name_map[pg], fvs)
+            
+            # Update buf_pg_profile
+            self.get_asic_buf_pg_profiles()
+
+            for pg in self.lossless_pgs:
+                self.app_db.wait_for_field_match("BUFFER_PG_TABLE", self.INTF + ":" + pg, {"profile": orig_lossless_profile})
+            for pg in self.pg_name_map:
+                fvs["SAI_INGRESS_PRIORITY_GROUP_ATTR_BUFFER_PROFILE"] = self.buf_pg_profile[pg]
+                self.asic_db.wait_for_field_match("ASIC_STATE:SAI_OBJECT_TYPE_INGRESS_PRIORITY_GROUP", self.pg_name_map[pg], fvs)
```

Comment:
remove. the zero cable len is one use case. Do not overload it. Add a separate test 
1. admin up a port. 
2. record existing profile [pg_lossless_<orig speed>_<cable len>]
3. change the port speed
4. verify new buffer profile is created - [pg_lossless_<new speed>_<cable len]
5. chk the buffer profiles attached to all the lossless pgs of that port - it should not be the orig profile. it should be the new profile

Reply:
Updated. The testcase `test_zero_cable_len_profile_update` is untouched now. I added another case `test_buffer_pg_update`. Thanks